vulnerabilities/models.py

```diff
@@ -642,6 +644,152 @@ def get_absolute_url(self):
         """
         return reverse("package_details", args=[self.purl])
 
+    # TODO: There are other methods, variables etc. in models.py that need similar renaming.
+    def get_fixed_by_package_versions(self, fix=True):
+        """
+        Return a queryset of all the package versions of this `package` that fix any vulnerability.
+        If `fix` is False, return all package versions whether or not they fix a vulnerability.
+        """
+        filter_dict = {
+            "name": self.name,
+            "namespace": self.namespace,
+            "type": self.type,
+            "qualifiers": self.qualifiers,
+            "subpath": self.subpath,
+        }
+
+        if fix:
+            filter_dict["packagerelatedvulnerability__fix"] = True
+
+        return Package.objects.filter(**filter_dict).distinct()
+
+    def sort_by_version(self, packages):
+        """
+        Return a list of `packages` sorted by version.
+        """
+        if not packages:
+            return []
+
+        version_class = RANGE_CLASS_BY_SCHEMES[packages[0].type].version_class
```

Comment:
Why not reuse the `version_class` property as defined under?

Reply:
Thank you @tdruez -- good catch, and you brought a smile to my face -- I'd created that new Package property method after I created the code in question and overlooked the fact the I could use my new method there.  ;-)  This works just fine:

```
    def sort_by_version(self, packages):
        """
        Return a list of `packages` sorted by version.
        """
        if not packages:
            return []

        # version_class = RANGE_CLASS_BY_SCHEMES[packages[0].type].version_class
        # return sorted(
        #     packages,
        #     key=lambda x: version_class(x.version),
        # )
        return sorted(
            packages,
            key=lambda x: self.version_class(x.version),
        )
```

Reply:
Done.